### PR TITLE
Add better error messages to the API

### PR DESCRIPTION
**Pull request by iHiD**
I've implemented this using Rails `I18n` functionality. Each error code that the API returns has a corresponding message in the I18n file.

I think this implements my bit of https://github.com/exercism/exercism/issues/4164#issuecomment-412716159
Closes https://github.com/exercism/exercism/issues/4159

### CODE REVIEW

File: app/controllers/api/solutions_controller.rb [hunk 1]
```diff
@@ -93,15 +93,13 @@ def latest
 
   def update
     begin
-      solution = SolutionBase.find_by_uuid!(params[:id], current_user.id)
+      solution = SolutionBase.find_by_uuid!(params[:id])
     rescue
-      # This covers both a non-existing solution and a solution
-      # belonging to someone else. We might want seperate messages
-      # but I'm choosing not to leak the existance of a solution
-      # at this stage, but unifying the errors.
```
Comment (iHiD) [inline app/controllers/api/solutions_controller.rb]: ^^ We're leaking this in other places anyway, so I've just removed this and am making the error messages consistent with the rest of the API

File: config/locales/en.yml [hunk 2]
```diff
@@ -30,4 +30,13 @@
 # available at http://guides.rubyonrails.org/i18n.html.
 
 en:
-  hello: "Hello world"
+  api:
+    errors:
+      solution_not_found: "This solution could not be found"
+      solution_not_accessible: "You do not have permission to view this solution"
+      track_not_found: "The track you specified does not exist"
+      track_ambiguous: "Please specify a track id"
```
Comment (kytrinyx) [inline config/locales/en.yml]: Can we make this `ID` instead of `id`?

**Comment by iHiD** [comment]
@kytrinyx I've updated this now. Note that there is not test-coverage for the `teams` path yet. I'll add that seperately.

File: config/locales/en.yml [hunk 3]
```diff
@@ -30,4 +30,15 @@
...
 en:
-  hello: "Hello world"
+  api:
+    errors:
+      duplicate_iteration: "No files you submitted have changed since your last iteration"
+      exercise_not_found: "The exercise you specified could not be found"
+      resource_not_found: "This URL was not recognised"
+      solution_not_found: "This solution could not be found"
+      solution_not_accessible: "You do not have permission to view this solution"
+      solution_not_unlocked: "You have not unlocked this solution"
```
[the remark this answers is not in the document]
Comment (iHiD) [reply config/locales/en.yml]: Agree. Updated it. Will merge/deploy when green.

**Pull request merged**